python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1553 @@
+            "extra_centroid_y": np.nan,
+            "intra_mag": np.nan,
+            "extra_mag": np.nan,
+            "intra_sn": np.nan,
+            "extra_sn": np.nan,
+            "intra_entropy": np.nan,
+            "extra_entropy": np.nan,
+            "intra_frac_bad_pix": np.nan,
+            "extra_frac_bad_pix": np.nan,
+            "intra_max_power_grad": np.nan,
+            "extra_max_power_grad": np.nan,
+            "intra_fx": np.nan,
+            "extra_fx": np.nan,
+            "intra_fy": np.nan,
+            "extra_fy": np.nan,
+        }
+
+        # Add Zernike coefficients
+        # Assert that zkCoeffAvg has the expected shape (1, nZernikes)
+        assert zkCoeffAvg.shape[0] == 1, (
+            f"Expected zkCoeffAvg to have shape (1, nZernikes), got {zkCoeffAvg.shape}"
+        )
+
+        for i, j in enumerate(self.config.nollIndices):
+            if i < zkCoeffAvg.shape[1]:  # Check against the number of coefficients
+                rowData[f"Z{j}"] = zkCoeffAvg[0, i] * u.um  # Access the scalar value
+            else:
+                rowData[f"Z{j}"] = 0.0 * u.um
+
+        zkTable.add_row(rowData)
+        self.log.debug("Added average row with %d Zernike terms", len(self.config.nollIndices))
+
+        # Add individual donut rows if we have donut stamps
+        # This creates one row per donut pair, showing individual donut
+        # contributions before AggregatorNet averaging
+        self.log.debug(
+            "Creating Zernike table: %d extra stamps, %d intra stamps, zkCoeffRaw shape: %s",
+            len(extraStamps),
+            len(intraStamps),
+            zkCoeffRaw.shape,
+        )
+
+        # Add individual donut rows
+        for i, (intraStamp, extraStamp) in enumerate(zip_longest(intraStamps, extraStamps)):
+            # Get the zernike coefficients for this donut
+            # TARTS provides individual coefficients per donut
+            if zkCoeffRaw.ndim == 2:
+                zk = zkCoeffRaw[i]  # Get the i-th row
+            else:
+                zk = zkCoeffRaw  # Already 1D
+
+            row: dict = dict()
+            row["label"] = f"donut{i + 1}"
+            row["used"] = True  # TARTS predictions are always used
+
+            # Add Zernike coefficients
+            for idx, nollIdx in enumerate(self.config.nollIndices):
+                if idx < len(zk):
+                    row[f"Z{nollIdx}"] = zk[idx] * u.um
+                else:
+                    row[f"Z{nollIdx}"] = 0.0 * u.um
+
+            # Get all metadata from stamps (already populated by
+            # createDonutStampsFromTarts)
+            for key in [
+                "MAG",
+                "SN",
+                "ENTROPY",
+                "FRAC_BAD_PIX",
+                "MAX_POWER_GRAD",
+                "FX",
+                "FY",
+                "CENT_X",
+                "CENT_Y",
+            ]:
+                if defocalType == "intra" and intraStamps:
```

Comment:
Also why access intraStamps here if you are in a for loop where each element it intraStamp?